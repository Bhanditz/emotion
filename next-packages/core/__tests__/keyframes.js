// @flow
/** @jsx jsx */
import { jsx, css, keyframes } from '@emotion/core'
import { safeQuerySelector } from 'test-utils'
import cases from 'jest-in-case'
import * as renderer from 'react-test-renderer'
import createCache from '@emotion/cache'
import { Provider } from '@emotion/core'

cases(
  'keyframes',
  opts => {
    safeQuerySelector('head').innerHTML = ''
    let cache = createCache()
    expect(
      renderer
        .create(<Provider value={cache}>{opts.render()}</Provider>)
        .toJSON()
    ).toMatchSnapshot()
    expect(cache.sheet.tags.map(tag => tag.textContent || '')).toMatchSnapshot()
  },
  {
    basic: {
      render: () => {
        // this isn't necessary and many people will likely use
        // keyframes as a tagged template literal but using a css call
        // means prettier will format it
        let animation = keyframes(css`
          from {
            color: green;
          }
          to {
            color: hotpink;
          }
        `)
        return (
          <div
            css={css`
              animation: ${animation};
            `}
          >
            {animation.name}
          </div>
        )
      }
    },
    'without css call': {
      render: () => {
        let animation = keyframes`
          from {
            color: green;
          }
          to {
            color: hotpink;
          }
        `
        return (
          <div
            css={css`
              animation: ${animation};
            `}
          >
            {animation.name}
          </div>
        )
      }
    },
    object: {
      render: () => {
        let animation = keyframes({
          from: {
            color: 'green'
          },
          to: {
            color: 'hotpink'
          }
        })
        let serialized = css({
          animation: `${animation} 1s`
        })
        console.log(serialized)

        return <div css={serialized}>{animation.name}</div>
      }
    }
  }
)
