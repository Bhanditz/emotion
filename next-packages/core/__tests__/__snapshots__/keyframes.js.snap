// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keyframes basic 1`] = `
<div
  className="css-11l88"
>
  animation-1bi8wtt
</div>
`;

exports[`keyframes basic 2`] = `
Array [
  ".css-11l88{-webkit-animation:animation-1bi8wtt;animation:animation-1bi8wtt;}",
  "@-webkit-keyframes animation-1bi8wtt{from{color:green;}to{color:hotpink;}}",
  "@keyframes animation-1bi8wtt{from{color:green;}to{color:hotpink;}}",
]
`;

exports[`keyframes multiple keyframes object 1`] = `
<div
  className="css-11ra1pv"
>
  animation-1bi8wtt
</div>
`;

exports[`keyframes multiple keyframes object 2`] = `
Array [
  ".css-11ra1pv{-webkit-animation:animation-1bi8wtt 1s animation-1ajb83a;animation:animation-1bi8wtt 1s animation-1ajb83a;}",
  "@-webkit-keyframes animation-1ajb83a{from{color:green;}to{color:yellow;}}",
  "@keyframes animation-1ajb83a{from{color:green;}to{color:yellow;}}",
  "@-webkit-keyframes animation-1bi8wtt{from{color:green;}to{color:hotpink;}}",
  "@keyframes animation-1bi8wtt{from{color:green;}to{color:hotpink;}}",
]
`;

exports[`keyframes object 1`] = `
<div
  className="css-1iju4zp"
>
  animation-o4c3es
</div>
`;

exports[`keyframes object 2`] = `
Array [
  ".css-1iju4zp{-webkit-animation:animation-o4c3es 1s;animation:animation-o4c3es 1s;}",
  "@-webkit-keyframes animation-o4c3es{from{color:green;}to{color:hotpink;}}",
  "@keyframes animation-o4c3es{from{color:green;}to{color:hotpink;}}",
]
`;

exports[`keyframes object with string keyframes 1`] = `
<div
  className="css-1mzl5he"
>
  animation-1bi8wtt
</div>
`;

exports[`keyframes object with string keyframes 2`] = `
Array [
  ".css-1mzl5he{-webkit-animation:animation-1bi8wtt 1s;animation:animation-1bi8wtt 1s;}",
  "@-webkit-keyframes animation-1bi8wtt{from{color:green;}to{color:hotpink;}}",
  "@keyframes animation-1bi8wtt{from{color:green;}to{color:hotpink;}}",
]
`;

exports[`keyframes without css call 1`] = `
<div
  className="css-11l88"
>
  animation-1bi8wtt
</div>
`;

exports[`keyframes without css call 2`] = `
Array [
  ".css-11l88{-webkit-animation:animation-1bi8wtt;animation:animation-1bi8wtt;}",
  "@-webkit-keyframes animation-1bi8wtt{from{color:green;}to{color:hotpink;}}",
  "@keyframes animation-1bi8wtt{from{color:green;}to{color:hotpink;}}",
]
`;
